### PR TITLE
update: allow up to 10 daysWorked

### PR DESCRIPTION
**Pull request by eisbilir**

### CODE REVIEW

File: src/services/WorkPeriodService.js [hunk 1]
```diff
@@ -273,9 +273,11 @@ async function updateWorkPeriod (currentUser, id, data) {
   if (_.isNil(thisWeek)) {
     throw new errors.ConflictError('Work Period dates are not compatible with Resource Booking dates')
   }
+  /* https://github.com/topcoder-platform/taas-apis/issues/428
   if (thisWeek.daysWorked < data.daysWorked) {
     throw new errors.BadRequestError(`Maximum allowed daysWorked is (${thisWeek.daysWorked})`)
   }
+  */
```
Comment (maxceem) [inline src/services/WorkPeriodService.js:280]: As we allow up to `10` and don't care about RB.endDate anymore, let's remove this code.

File: src/services/ResourceBookingService.js [hunk 2]
```diff
@@ -260,6 +261,7 @@ async function _ensurePaidWorkPeriodsNotDeleted (resourceBookingId, oldValue, ne
       throw new errors.ConflictError(`Cannot make maximum daysWorked (${newWP.daysWorked}) to the value less than daysPaid (${wp.daysPaid}) for WorkPeriod: ${wp.id}`)
     }
   })
+  */
```
Comment (maxceem) [inline src/services/ResourceBookingService.js:264]: I'm not sure why this check is not valid anymore?
It looks like if we allow maximum 10 working days, we still should not allow this value to be less than `daysPaid`.

**Review by maxceem — COMMENTED** [review]
@eisbilir it works good. 

Though I'm not sure about changes for Resource Bookings logic, could you please check the comments below?

File: src/services/ResourceBookingService.js (hunk 2 above)
Comment (eisbilir) [reply src/services/ResourceBookingService.js:264]: this checks if `daysPaid` exceeds maximum possible `daysWorked`.
maximum possible `daysWorked` is calculated regarding the startDay and endDate.
this check won't let us update any RB which has a WP has more paid daysWorked than expected (1-5).
So I moved the part of the logic we need to eventHandler, I don't allow to lower `daysWorked` less then `daysPaid` regardless of dates here:
https://github.com/eisbilir/taas-apis/blob/cb194d63722be0d8b6e1b040172c3236cfeadcd0/src/eventHandlers/ResourceBookingEventHandler.js#L192
https://github.com/eisbilir/taas-apis/blob/cb194d63722be0d8b6e1b040172c3236cfeadcd0/src/eventHandlers/ResourceBookingEventHandler.js#L197
https://github.com/eisbilir/taas-apis/blob/cb194d63722be0d8b6e1b040172c3236cfeadcd0/src/eventHandlers/ResourceBookingEventHandler.js#L207
https://github.com/eisbilir/taas-apis/blob/cb194d63722be0d8b6e1b040172c3236cfeadcd0/src/eventHandlers/ResourceBookingEventHandler.js#L212

Example 1
WP-1 has 4 daysWorked as maximum.
daysWorked was increased to 7.
payment was processed and daysPaid increased to 2.
RB dates updated and it forced WP-1 to have 3 daysWorked.
I will allow both RB dates and WP-1 daysWorked (3) to be updated.

Example 2
WP-1 has 4 daysWorked as maximum.
daysWorked was increased to 7.
payment was processed and daysPaid increased to 6.
RB dates updated and it forced WP-1 to have 3 daysWorked.
I will allow RB dates to be updated but WP-1 will keep it daysWorked at 7, same as before update.

WP-1 refers to 1st week.
we use recalculating of daysWorked logic for only the first and the last WP.

Comment (maxceem) [reply src/services/ResourceBookingService.js:264]: Thanks, got it.

**Review by maxceem — APPROVED** [review]
@eisbilir works good as per local testing after updating ES.

**Pull request merged**